shard-operator-parent/shard-operator-v2/src/main/java/com/redhat/service/smartevents/shard/operator/v2/controllers/ManagedProcessorController.java

```diff
@@ -0,0 +1,171 @@
+import io.javaoperatorsdk.operator.api.reconciler.Reconciler;
+import io.javaoperatorsdk.operator.api.reconciler.RetryInfo;
+import io.javaoperatorsdk.operator.api.reconciler.UpdateControl;
+import io.javaoperatorsdk.operator.processing.event.source.EventSource;
+
+@ApplicationScoped
+@ControllerConfiguration(labelSelector = LabelsBuilder.V2_RECONCILER_LABEL_SELECTOR)
+public class ManagedProcessorController implements Reconciler<ManagedProcessor>,
+        EventSourceInitializer<ManagedProcessor>, ErrorStatusHandler<ManagedProcessor> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ManagedProcessorController.class);
+
+    @ConfigProperty(name = "event-bridge.executor.deployment.timeout-seconds")
+    int executorTimeoutSeconds;
+
+    @ConfigProperty(name = "event-bridge.executor.poll-interval.milliseconds")
+    int executorPollIntervalMilliseconds;
+
+    @Inject
+    KubernetesClient kubernetesClient;
+
+    @Inject
+    ManagedProcessorService managedProcessorService;
+
+    @Inject
+    BridgeErrorHelper bridgeErrorHelper;
+
+    @Inject
+    OperatorMetricsService metricsService;
+
+    @Override
+    public List<EventSource> prepareEventSources(EventSourceContext<ManagedProcessor> eventSourceContext) {
+
+        List<EventSource> eventSources = new ArrayList<>();
+        eventSources.add(EventSourceFactory.buildResourceInformer(kubernetesClient, LabelsBuilder.V2_OPERATOR_NAME,
+                ManagedProcessor.COMPONENT_NAME,
+                CamelIntegration.class));
+
+        return eventSources;
+    }
+
+    @Override
+    public UpdateControl<ManagedProcessor> reconcile(ManagedProcessor managedProcessor, Context context) {
+        String managedProcessorName = managedProcessor.getMetadata().getName();
+        String managedProcessorNamespace = managedProcessor.getMetadata().getNamespace();
+
+        LOGGER.info("Reconciling ManagedProcessor: '{}' in namespace '{}'",
+                managedProcessorName,
+                managedProcessorNamespace);
+
+        ManagedProcessorStatus processorStatus = managedProcessor.getStatus();
+
+        if (!processorStatus.isReady() && isTimedOut(processorStatus)) {
+            // notifyManagerOfFailure
+            processorStatus.markConditionFalse(ConditionTypeConstants.READY);
+            return UpdateControl.updateStatus(managedProcessor);
+        }
+
+        String integrationName = String.format("integration-%s", managedProcessorName);
+        CamelIntegration camelIntegration = managedProcessorService.fetchOrCreateCamelIntegration(managedProcessor, integrationName);
+
+        if (!camelIntegration.isReady()) {
+            LOGGER.info("CamelIntegration for the ManagedProcessor '{}' in namespace '{}' is not ready",
+                    managedProcessorName,
+                    managedProcessorNamespace);
+            if (!processorStatus.isConditionTypeFalse(ConditionTypeConstants.READY)) {
```

Comment:
@lucamolteni Should this be setting the `ManagedProcessorStatus.CAMEL_INTEGRATION_AVAILABLE` `Condition` to `false` instead?

Reply:
We've been trying to keep `Condition` constants [here](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/blob/main/infra-parent/infra-v2/src/main/java/com/redhat/service/smartevents/infra/v2/api/models/DefaultConditions.java) so they're reusable from both the `manager` and `operator`.

Reply:
it's using `markConditionFalse` am I missing something

Reply:
@lucamolteni Currently you're setting the `Ready` condition to `false` at this point. I think it makes more sense to set the `ManagedProcessorStatus.CAMEL_INTEGRATION_AVAILABLE` to `false` as it makes it clearer to us what we're still waiting on and you then set the `ManagedProcessorStatus.CAMEL_INTEGRATION_AVAILABLE` to `true` immediately after this part of the code.